ucm2/Intel/sof-hda-dsp/HiFi-analog-sof.conf

```diff
@@ -0,0 +1,492 @@
+# Generic HDA devices for analog I/O
+
+Define.FrontMicName ""
+Define.FrontMicJack ""
+Define.DeviceFrontMic ""
+Define.LineDevice ""
+Define.hpvol "Headphone"
+Define.hpjack "Headphone Jack"
+Define.loctl "Line"
+Define.lovol "Line"
+
+# IPC version
+# TODO need to duplicate below definions to True and False since Define after
+# this if is not seeing variable SOFIPCVer.
+If.SOFIPCVer {
+	Condition {
+		Type ControlExists
+		Control "name='Post Mixer Analog Playback IIR Eq bytes'"
+	}
+	True.Define {
+	      SOFIPCVer "ipc4"
+	      PostMixerAnalogPlaybackIIRBytes "Post Mixer Analog Playback IIR Eq bytes"
+	      PostMixerAnalogPlaybackFIRBytes "Post Mixer Analog Playback FIR Eq bytes"
+	      PostMixerAnalogPlaybackDRCBytes "Post Mixer Analog Playback DRC bytes"
+	      PostMixerAnalogPlaybackDRCSwitch "Post Mixer Analog Playback DRC switch"
+	      SpeakerIirBlob "${ConfTopDir}/sof/${var:SOFIPCVer}/eq_iir/highpass_100hz_0db_48khz.blob"
+	      SpeakerFirBlob "${ConfTopDir}/sof/${var:SOFIPCVer}/eq_fir/pass.blob"
+	      SpeakerDrcBlob "${ConfTopDir}/sof/${var:SOFIPCVer}/drc/speaker_default.blob"
+	      HeadphoneIirBlob "${ConfTopDir}/sof/${var:SOFIPCVer}/eq_iir/pass.blob"
+	      HeadphoneFirBlob "${ConfTopDir}/sof/${var:SOFIPCVer}/eq_fir/pass.blob"
+	      HeadphoneDrcBlob "${ConfTopDir}/sof/${var:SOFIPCVer}/drc/passthrough.blob"
+	}
+	False.Define {
+	      SOFIPCVer "ipc3"
+	      PostMixerAnalogPlaybackIIRBytes "EQIIR1.0 eqiir_coef_1"
+	      PostMixerAnalogPlaybackFIRBytes "EQFIR1.0 eqfir_coef_1"
+	      PostMixerAnalogPlaybackDRCBytes "not available"
+	      PostMixerAnalogPlaybackDRCSwitch "not available"
+	      SpeakerIirBlob "${ConfTopDir}/sof/${var:SOFIPCVer}/eq_iir/highpass_100hz_0db_48khz.blob"
+	      SpeakerFirBlob "${ConfTopDir}/sof/${var:SOFIPCVer}/eq_fir/pass.blob"
+	      SpeakerDrcBlob "${ConfTopDir}/sof/${var:SOFIPCVer}/drc/speaker_default.blob"
+	      HeadphoneIirBlob "${ConfTopDir}/sof/${var:SOFIPCVer}/eq_iir/pass.blob"
+	      HeadphoneFirBlob "${ConfTopDir}/sof/${var:SOFIPCVer}/eq_fir/pass.blob"
+	      HeadphoneDrcBlob "${ConfTopDir}/sof/${var:SOFIPCVer}/drc/passthrough.blob"
+	}
+}
+
+# Include blob customizations: sys_vendor / product_name .conf
```

Comment:
I don't understand the "Include blob customizations', how does this override the defaults?

Reply:
The file ProductConfigSOF, it it exists, can re-define the previously defined default blobs, e.g. SpeakerFirBlob to enable the FIR filter that is by default passthrough. Or the SpeakerIirBlob that is set for default 100 Hz high-pass.

Reply:
Isn't it simpler conceptually to do

if (product config)
    include product_config_file
else 
    include default_config_file

It's very hard to debug if parts of the config are overridden and others are not. Only experts would understand what they are doing, the rest of the world needs things to be error-proof.

Reply:
Yes it could be like that. The benefit of current way is that the product config does not need to define every variable, only the ones that should change. E.g. headphone settings should be changed from pass-through default to something else only by user.

With the if - else you propose few code lines would be saved but there would be risk for UCM silently not working if product config misses to set some variables.